autogluon/utils/tabular/features/utils.py

```diff
@@ -1,49 +1,22 @@
 import logging
-from collections import defaultdict
 
 import numpy as np
+from pandas import DataFrame, Series
 
 logger = logging.getLogger(__name__)
 
 
-def get_type_family(dtype):
-    """From dtype, gets the dtype family."""
-    try:
-        if dtype.name is 'category':
-            return 'category'
-        if 'datetime' in dtype.name:
-            return 'datetime'
-        elif np.issubdtype(dtype, np.integer):
-            return 'int'
-        elif np.issubdtype(dtype, np.floating):
-            return 'float'
-    except Exception as err:
-        logger.exception(f'Warning: dtype {dtype} is not recognized as a valid dtype by numpy! AutoGluon may incorrectly handle this feature...')
-        logger.exception(err)
+# TODO: Add documentation
+def clip_and_astype(df: DataFrame, columns: list = None, clip_min=0, clip_max=255, dtype: str = 'uint8') -> DataFrame:
+    if columns is None:
+        df = np.clip(df, clip_min, clip_max).astype(dtype)
+    elif columns:
+        df[columns] = np.clip(df[columns], clip_min, clip_max).astype(dtype)
+    return df
```

Comment:
Take into account, that there is a risk of data corruption with underflows:
```
In[1]: np.float32(20140131.0)
Out[1]: 20140132.0
```
Sometimes it's quite bad:
```
In[1]: np.int(4.99999)
Out[1]: 4

```
This is not always bad: for instance, in case of stacker, this compression is adding some noise and have regularizing effect.

Why not re-scale before clipping?

Reply:
This clipping logic is generally reserved for dtype byte reduction when it is lossless during training. Such as an int64 feature only containing values between 0 and 200 could be converted to uint8 losslessly. The clipping part is at inference time when the values could be -9999 and 9999 to ensure they are set to 0 and 255. Normalization / re-scaling is something not tackled in this PR, as it was not existing in prior feature generator functionality.